torch/testing/_internal/opinfo_helper.py

```diff
@@ -0,0 +1,120 @@
+EXTENSIBLE_DTYPE_DISPATCH = (
+    all_types_and_complex_and,
+    floating_types_and,
+    floating_and_complex_types_and,
+    integral_types_and,
+    all_types_and,
+)
+
+# Better way to acquire devices?
+DEVICES = ['cpu'] + (['cuda'] if TEST_CUDA else [])
+
+# Class to tag the dynamically generated types.
+
+
+class _dynamic_dispatch_dtypes(_dispatch_dtypes):
+    pass
+
+
+def get_supported_dtypes(op, sample_inputs_fn, device):
+    # Returns the supported dtypes for the given operator and device pair.
+    assert device in ['cpu', 'cuda']
+    if not TEST_CUDA and device == 'cuda':
+        warnings.warn("WARNING: CUDA is not available, information pertaining to CUDA dtypes could be wrong")
+
+    supported_dtypes = set()
+    for dtype in all_types_and_complex_and(torch.bool, torch.bfloat16, torch.half):
+        samples = sample_inputs_fn(op, device, dtype, False)
+        for sample in samples:
+            try:
+                op(sample.input, *sample.args, **sample.kwargs)
+                supported_dtypes.add(dtype)
+            except RuntimeError as re:
+                if "not implemented for" not in str(re):
+                    raise re
+                pass
+
+    return _dynamic_dispatch_dtypes(supported_dtypes)
+
+
+def dtypes_dispatch_hint(dtypes):
+    return_type = collections.namedtuple('dispatch_hint_return_type', 'dispatch_fn dispatch_fn_str')
+
+    # Find the closest dispatcher from the set of available dispatchers
+
+    # Metric for finding the closest dispatcher
+    # Intersection Over Union
+    def iou(dispatch_dtypes):
+        return len(set(dtypes).intersection(dispatch_dtypes)) * 1.0 / len(set(dtypes).union(dispatch_dtypes))
+
+    dispatch_iou_score = {dispatch: iou(dispatch()) for dispatch in COMPLETE_DTYPES_DISPATCH}
+
+    dispatch, score = max(dispatch_iou_score.items(), key=lambda x: x[1])
+
+    if dispatch in COMPLETE_DTYPES_DISPATCH and score == 1.0:
+        return return_type(dispatch, dispatch.__name__ + "()")
+
+    dispatch_iou_score = {dispatch: iou(dispatch()) for dispatch in EXTENSIBLE_DTYPE_DISPATCH}
+    dispatch, _ = max(dispatch_iou_score.items(), key=lambda x: x[1])
+
+    return return_type(partial(dispatch, *tuple(set(dtypes) - set(dispatch()))),
+                       dispatch.__name__ + str(tuple(set(dtypes) - set(dispatch()))))
+
+
+def is_dynamic_dtype_set(op):
+    return op.dynamic_dtypes
+
+
+def str_format_dynamic_dtype(op):
+    fmt_str = """
+        OpInfo({name},
+               dtypes={dtypesIfCPU},
+               dtypesIfCUDA={dtypesIfCUDA},
+        )
+        """.format(name=op.name,
+                   dtypesIfCPU=dtypes_dispatch_hint(op.dtypesIfCPU).dispatch_fn_str,
+                   dtypesIfCUDA=dtypes_dispatch_hint(op.dtypesIfCUDA).dispatch_fn_str)
+
+    return fmt_str
+
+
+# Run using `python -m torch.testing._internal.opinfo_helper`
+if __name__ == '__main__':
+    filtered_ops = list(filter(is_dynamic_dtype_set, cmi.op_db))
+
+    for op in filtered_ops:
```

Comment:
Let's contextualize this printing with a string explaining why each of these opinfos is being printed